src/axolotl/utils/config/models/internals/__init__.py

```diff
@@ -12,3 +12,9 @@ class GPUCapabilities(BaseModel):
     n_gpu: int = Field(default=1)
     n_node: int = Field(default=1)
     compute_capability: Optional[str] = Field(default=None)
+
+
+class EnvCapabilities(BaseModel):
+    """model to manage the environment capabilities statically"""
+
+    torch_version: Optional[str] = Field(default=None)
```

Comment:
@winglian , this could be an alternative to the `def flash_attn` you created in the Unsloth PR and allow re-use of import checks across modules. This can also include bnb / other stuff later to check before we import to allow optional packages in future.

Reply:
good call-out. probably out of scope for this PR, but we can upgrade the others in follow-up PRs.